python/src/robyn/calibration/media_effect_calibration.py

```diff
@@ -0,0 +1,206 @@
+        # Calculate MAPE using the totals
+        mape = np.abs((total_pred_effect - lift_value) / lift_value)
+        return float(mape)
+
+    def _calculate_total_effect_score(
+        self, predictions: pd.Series, lift_value: float, spend: float, channel: str
+    ) -> float:
+        """
+        Calculates calibration score for total effects including carryover.
+
+        Args:
+            predictions: Series of predictions to evaluate
+            lift_value: Actual lift value to compare against
+            spend: Spend amount for the channel
+            channel: Channel name for getting correct hyperparameters
+        """
+        pred_effect = self.media_transformation.apply_media_transforms(predictions, channel)
+        # Get total effect including carryover
+        total_effect = self.media_transformation.apply_carryover_effect(pred_effect)
+        # Calculate MAPE using the totals
+        mape = np.abs((total_effect - lift_value) / lift_value)
+        return float(mape)
+
+    def calibrate(self) -> CalibrationResult:
+        """
+        Performs calibration by comparing model predictions to experimental results.
+        Returns CalibrationResult with scores per channel.
+        """
+        calibration_scores = {}
+
+        for channel, data in self.calibration_input.channel_data.items():
+            # Handle single or combined channels
+            channel_list = self._split_channel(channel)
+
+            try:
+                # Get predictions for all relevant channels
+                predictions = []
+                for ch in channel_list:
+                    pred = self._get_channel_predictions(ch, data)
+                    predictions.append(pred)
+
+                # Sum predictions for combined channels
+                total_pred = pd.concat(predictions, axis=1).sum(axis=1)
+
+                # Use the first channel's hyperparameters for combined channels
+                channel_for_params = channel_list[0]
+
+                # Calculate calibration score based on scope
+                if data.calibration_scope == CalibrationScope.IMMEDIATE:
+                    score = self._calculate_immediate_effect_score(
+                        total_pred, data.lift_abs, data.spend, channel_for_params
+                    )
+                else:
+                    score = self._calculate_total_effect_score(
+                        total_pred, data.lift_abs, data.spend, channel_for_params
+                    )
+
+                calibration_scores[channel] = score
+
+            except Exception as e:
+                print(f"Error calculating calibration for channel {channel}: {str(e)}")
+                # You might want to handle this differently based on your requirements
+                calibration_scores[channel] = float("inf")
+
+        return CalibrationResult(
+            channel_scores=calibration_scores,
+            calibration_constraint=0.05,  # Default constraint
```

Comment:
This seems to be hard coded. CalibrationResult definition would be right place to have default value.